Core/src/org/sleuthkit/autopsy/contentviewers/ContextViewer.java

```diff
@@ -0,0 +1,427 @@
+            .addGroup(layout.createSequentialGroup()
+                .addContainerGap()
+                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
+                    .addGroup(layout.createSequentialGroup()
+                        .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
+                            .addComponent(jLabel1)
+                            .addGroup(layout.createSequentialGroup()
+                                .addGap(6, 6, 6)
+                                .addComponent(jSourceNameLabel)
+                                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                                .addComponent(jSourceTextLabel, javax.swing.GroupLayout.DEFAULT_SIZE, 192, Short.MAX_VALUE)))
+                        .addGap(36, 36, 36))
+                    .addGroup(layout.createSequentialGroup()
+                        .addComponent(jSourceGoToResultButton)
+                        .addContainerGap(javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))))
+        );
+        layout.setVerticalGroup(
+            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
+            .addGroup(layout.createSequentialGroup()
+                .addContainerGap()
+                .addComponent(jLabel1)
+                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
+                    .addComponent(jSourceNameLabel)
+                    .addComponent(jSourceTextLabel))
+                .addGap(18, 18, 18)
+                .addComponent(jSourceGoToResultButton)
+                .addGap(0, 203, Short.MAX_VALUE))
+        );
+    }// </editor-fold>//GEN-END:initComponents
+
+    private void jSourceGoToResultButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jSourceGoToResultButtonActionPerformed
+
+        final DirectoryTreeTopComponent dtc = DirectoryTreeTopComponent.findInstance();
+
+        // Navigate to the source context artifact.
+        if (sourceContextArtifact != null) {
+            dtc.viewArtifact(sourceContextArtifact);
+        }
+
+    }//GEN-LAST:event_jSourceGoToResultButtonActionPerformed
+
+    @Override
+    public void setNode(Node selectedNode) {
+        if ((selectedNode == null) || (!isSupported(selectedNode))) {
+            resetComponent();
+            return;
+        }
+
+        AbstractFile file = selectedNode.getLookup().lookup(AbstractFile.class);
+        try {
+            populateSourceContextData(file);
+        } catch (NoCurrentCaseException | TskCoreException ex) {
+            logger.log(Level.SEVERE, "Exception displaying context for file {0}", file); //NON-NLS
+        }
+    }
+
+    @Override
+    public String getTitle() {
+        return Bundle.ContextViewer_title();
```

Comment:
@rcordovano mentioned to me recently that it is preferred that  messages are defined just above the function they are used instead of all at the top of the class.